### PR TITLE
[WIP] Implement R2D2

### PR DESCRIPTION
**Pull request by alvkao58**
## What do these changes do?

Adds LSTM functionality to DQN, implements R2D2.

## Related issue number

#3148

### CODE REVIEW

File: python/ray/rllib/agents/dqn/dqn_policy_graph.py [hunk 1]
```diff
@@ -380,17 +400,34 @@ def __init__(self, observation_space, action_space, config):
...
             observation_space,
             action_space,
             self.sess,
-            obs_input=self.cur_observations,
+            obs_input=self.obs_t,
             action_sampler=self.output_actions,
             loss=model.loss() + self.loss.loss,
             loss_inputs=self.loss_inputs,
+            state_inputs=model.state_in + model2.state_in,
+            state_outputs=model.state_out + model2.state_out,
```
Comment (ericl) [inline python/ray/rllib/agents/dqn/dqn_policy_graph.py]: I think you can pass in the state tensors for model1 to the get_model() for model2. That way, they will share the same set of input placeholders for RNN state.

**Comment by theSoenke** [comment]
@alvkao58 What is the status on this? Does this implementation already works?

**Comment by richardliaw** [comment]
cc @joneswong, as we talked earlier today

**Comment by joneswong** [comment]
OK, I can take this after the Retreat.

**Pull request closed without merging**